cpp/src/parquet/page_index.cc

```diff
@@ -725,6 +761,29 @@ class PageIndexBuilderImpl final : public PageIndexBuilder {
     }
   }
 
+  template <typename Builder>
```

Comment:
Instead of templating this on the builder type, why wouldn't you simply pass the `module_type` as a function argument?